pkg/network/ebpf/c/protocols/kafka/kafka-parsing.h

```diff
@@ -1620,6 +1622,16 @@ static __always_inline bool kafka_process(conn_tuple_t *tup, kafka_info_t *kafka
 
     log_debug("kafka: kafka_header.api_key: %d api_version: %d", kafka_header.api_key, kafka_header.api_version);
 
+    // Report api version hits telemetry
+    switch (kafka_header.api_key) {
+        case KAFKA_PRODUCE:
+            update_classified_produce_api_version_hits_telemetry(kafka_tel, kafka_header.api_version);
+            break;
+        case KAFKA_FETCH:
+            update_classified_fetch_api_version_hits_telemetry(kafka_tel, kafka_header.api_version);
+            break;
+    }
```

Comment:
I'd move that after the condition in line 1635, to ensure we're dealing with valid kafka requests

Reply:
The verifier complains about `update_classified_fetch_api_version_hits_telemetry` if I move it :( it shouldn't have any effect because it is being checked before getting here (IIUC)

Reply:
> it shouldn't have any effect because it is being checked before getting here (IIUC)

Where? the condition that should verify the traffic represents a valid kafka request is in the condition after this code

> The verifier complains about update_classified_fetch_api_version_hits_telemetry if I move it

What does it complain about? we should be able to address it

Reply:
I have no idea what's going on but it works now
and I meant `is_valid_kafka_request_header` is called both in the classification and again at the start of the decoding so I think the one in the decoding is obsolete